contracts/CharityChallenge.sol

```diff
@@ -141,22 +144,22 @@ contract CharityChallenge {
                 uint length = npoAddresses.length;
                 uint256 donatedAmount = 0;
                 if (makerFee > 0) {
-                    uint256 amount = totalContractBalance * makerFee / 100;
-                    donatedAmount += amount;
+                    uint256 amount = totalContractBalance.mul(makerFee).div(10000);
+                    donatedAmount = donatedAmount.add(amount);
                     contractOwner.transfer(amount);
                     emit Fee(contractOwner, amount);
                 }
-                uint256 amountForNPO = totalContractBalance - donatedAmount;
+                uint256 amountForNPO = totalContractBalance.sub(donatedAmount);
                 for (uint i = 0; i < length - 1; i++) {
                     address payable npo = npoAddresses[i];
                     uint8 ratio = npoRatios[npo];
-                    uint256 amount = amountForNPO * ratio / sumRatio;
-                    donatedAmount += amount;
+                    uint256 amount = amountForNPO.mul(ratio).div(sumRatio);
+                    donatedAmount = donatedAmount.add(amount);
                     npo.transfer(amount);
                     emit Donated(npo, amount);
                 }
                 // Don't want to keep any amount in the contract
-                uint256 remainingAmount = totalContractBalance - donatedAmount;
+                uint256 remainingAmount = totalContractBalance.sub(donatedAmount);
```

Comment:
* is there a chance that `donatedAmount` is greater than `totalContractBalance` (due to precision through calculations above)? If not, it's ok to leave as is. If yes, would the current code statements or the next statements throw error? 

* can we use `address(this).balance` to know the remaining balance here without doing calculation, as the transfer statements have been already executed above? 

* if we can use `address(this).balance` after all the transfer statements, i guess we don't really need to have variable `donatedAmount` for bookkeeping.

Reply:
I've tried to use `address(this).balance` here and it breaks the tests. I suppose that the balance of the contract is updated only at the end of the function call.

Reply:
Yep, it simplified the code greatly, thanks!